src/App.tsx

```diff
@@ -1,26 +1,28 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { ApolloProvider } from '@apollo/client';
+import { BrowserRouter, Routes, Route} from 'react-router-dom';
+import { client } from './config/apollo';
+import { AppContextProvider } from './context/appContext';
+import CarForm from './pages/CarForm';
+import Dashboard from './pages/Dashboard';
+import Home from './pages/Home';
+import CarDetails from './pages/CarDetails';
 
-function App() {
+const App = () => {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    <ApolloProvider client={client}>
+      <AppContextProvider>
+        <BrowserRouter>
+          <Routes>
+            <Route path='/' element={<Home/>}/>
+            <Route path='/dashboard' element={<Dashboard/>}/>
+            <Route path='/car-form' element={<CarForm/>}/>
+            <Route path='/favorites' element={<Dashboard/>}/>
```

Comment:
the logic for Dashboard and Favorites is different so I think it would be better if you create a different component for favorites instead of forcing the Dashboard component to handle everything